pkgs/development/libraries/science/math/lcalc/default.nix

```diff
@@ -0,0 +1,78 @@
+{ stdenv
+, fetchurl
+, fetchpatch
+, pari
+}:
+
+stdenv.mkDerivation rec {
+  version = "1.23";
+  pname = "lcalc";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    # original at http://oto.math.uwaterloo.ca/~mrubinst/L_function_public/CODE/L-${version}.tar.gz, no longer available
+    # "newer" version at google code https://code.google.com/archive/p/l-calc/source/default/source
+    url = "http://mirrors.mit.edu/sage/spkg/upstream/lcalc/lcalc-${version}.tar.bz2";
+    sha256 = "1c6dsdshgxhqppjxvxhp8yhpxaqvnz3d1mlh26r571gkq8z2bm43";
+  };
+
+  preConfigure = "cd src";
```

Comment:
Please use `sourceRoot`: https://nixos.org/nixpkgs/manual/#ssec-unpack-phase

Reply:
The patches assume to be in the `lcalc-${version}` directory. Using source root, it'd be necessary to

- hard code the name of that directory (it would be way nicer if the standard `unpackPhase` would rename that directory to a standardized name, like `source`)
- add `stripLen` or however that option was called to all patches

I think its nicer the way it is.